tests/compatibility_tests/add_or_replace_documents/test_add_documents.py

```diff
@@ -0,0 +1,100 @@
+import pytest
+from tests.compatibility_tests.base_test_case.base_compatibility_test import BaseCompatibilityTestCase
+
+@pytest.mark.marqo_version('2.0.0')
+class TestAddDocuments(BaseCompatibilityTestCase):
+    structured_index_name = "test_add_doc_api_structured_index"
+    unstructured_index_name = "test_add_doc_api_unstructured_index"
+
+    indexes_to_test_on = [{
+        "indexName": structured_index_name,
+        "type": "structured",
+        "model": "sentence-transformers/all-MiniLM-L6-v2",
+        "normalizeEmbeddings": False,
+        "allFields": [
+            {"name": "Title", "type": "text"},
+            {"name": "Description", "type": "text"},
+            {"name": "Genre", "type": "text"},
+        ],
+        "tensorFields": ["Title", "Description", "Genre"],
+    },
+        {
+        "indexName": unstructured_index_name,
+        "type": "unstructured",
+        "model": "sentence-transformers/all-MiniLM-L6-v2",
+        "normalizeEmbeddings": False,
+    }]
+
+    text_docs = [{
+        "Title": "The Travels of Marco Polo",
+        "Description": "A 13th-century travelogue describing the travels of Polo",
+        "Genre": "History",
+        "_id": "article_602"
+    },
+    {
+        "Title": "Extravehicular Mobility Unit (EMU)",
+        "Description": "The EMU is a spacesuit that provides environmental protection",
+        "_id": "article_591",
+        "Genre": "Science"
+    }]
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.indexes_to_delete = [index['indexName'] for index in cls.indexes_to_test_on]
+        super().tearDownClass()
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+
+    def prepare(self):
+        self.logger.info(f"Creating indexes {self.indexes_to_test_on} in test case: {self.__class__.__name__}")
+        self.create_indexes(self.indexes_to_test_on)
+
+        try:
+            self.logger.debug(f'Feeding documents to {self.indexes_to_test_on}')
+            for index in self.indexes_to_test_on:
+                if index.get("type") is not None and index.get('type') == 'structured':
```

Comment:
Subtests alone won't work here, because they just can be a wrapper above assertionErrors. Any other types of exceptions / errors and the subtest will fail and make the whole test fail. So I'm now using a try catch block + a subtest. Try catch block will be useful for any generic errors where as assertion errors can be logged by subtests

Reply:
I'm not sure if you should worry about bugs in the test. It's fine for the whole test to fail if the problem is not an assertion. Try/catch to handle things that shouldn't be happening risks masking bugs

Reply:
The test will eventually fail, this just enables it to run through all the test cases that are being iterated over using a for loop. If an exception happens maybe due to a bug, it will just be collected, and reported at the end. The logic here is such that if there's an exception we just report it at the end and fail the test.